src/components/menuAppBar/NavButton.tsx

```diff
@@ -0,0 +1,25 @@
+import React, { useState, useEffect } from "react";
+import { useRouter } from "next/router";
+
+const NavButton = props => {
+  const router = useRouter();
+  const [isActive, setIsActive] = useState(false);
```

Comment:
This might be redundant. This component might be stateless. All  you need to do is check if its active but just as const variable. This component will be re-rendered due to parent re-render